client/utils/utils.go

```diff
@@ -49,12 +59,70 @@ func SendTx(txCtx authctx.TxContext, cliCtx context.CLIContext, msgs []sdk.Msg)
 		return err
 	}
 
+	txCtx, err = enrichCtxWithGasIfGasAuto(txCtx, cliCtx, passphrase, msgs)
+	if err != nil {
+		return err
+	}
+
 	// build and sign the transaction
 	txBytes, err := txCtx.BuildAndSign(cliCtx.FromAddressName, passphrase, msgs)
 	if err != nil {
 		return err
 	}
-
 	// broadcast to a Tendermint node
 	return cliCtx.EnsureBroadcastTx(txBytes)
 }
+
+func enrichCtxWithGasIfGasAuto(txCtx authctx.TxContext, cliCtx context.CLIContext, passphrase string, msgs []sdk.Msg) (authctx.TxContext, error) {
+	if cliCtx.Gas == 0 {
```

Comment:
Might be slightly clearer to put this conditional outside of the function, and only call the function if `cliCtx.Gas == 0`

Reply:
As long as linter doesn't complain yes, that was my intent